packages/big-design/src/components/Table/Item/Item.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { memo, useContext } from 'react';
+
+import { Checkbox } from '../../Checkbox';
+import { TableSectionContext } from '../context';
+import { Cell } from '../Cell/Cell';
+
+import { StyledTableItem } from './styled';
+
+export interface ItemProps extends React.TableHTMLAttributes<HTMLTableRowElement> {
+  isSelectable: boolean;
+  selected?: boolean;
+  onItemSelect?(nextValue: boolean): void;
+}
+
+export const Item: React.FC<ItemProps> = memo(({ children, isSelectable, selected = false, onItemSelect }) => {
+  const tableSectionContext = useContext(TableSectionContext);
+
+  const handleSelect = () => {
+    if (typeof onItemSelect === 'function') {
+      onItemSelect(!selected);
+    }
```

Comment:
In this case I'm doing it since we don't need to return anything. It has an implicit return type of `void`. If we change it to `return typeof onItemSelect === 'function' && onItemSelect(!selected)` then it would return a boolean | whatever `onItemSelect` returns (in this case void). 

We use it it in our components jsx a ton because if you return `false` react will just not render anything from it.